src/interp/interp-math.h

```diff
@@ -64,32 +64,107 @@ T WABT_VECTORCALL CanonNaN(T val) {
-template <typename T> T WABT_VECTORCALL IntXor(T lhs, T rhs) { return lhs ^ rhs; }
-template <typename T> T WABT_VECTORCALL IntShl(T lhs, T rhs) { return lhs << ShiftMask(rhs); }
-template <typename T> T WABT_VECTORCALL IntShr(T lhs, T rhs) { return lhs >> ShiftMask(rhs); }
-template <typename T> T WABT_VECTORCALL IntMin(T lhs, T rhs) { return std::min(lhs, rhs); }
-template <typename T> T WABT_VECTORCALL IntMax(T lhs, T rhs) { return std::max(lhs, rhs); }
-template <typename T> T WABT_VECTORCALL IntAndNot(T lhs, T rhs) { return lhs & ~rhs; }
-template <typename T> T WABT_VECTORCALL IntAvgr(T lhs, T rhs) { return (lhs + rhs + 1) / 2; }
-template <typename T> T WABT_VECTORCALL Xchg(T lhs, T rhs) { return rhs; }
+template <typename T>
+T ShiftMask(T val) {
+  return val & (sizeof(T) * 8 - 1);
+}
+
+template <typename T>
+bool WABT_VECTORCALL IntEqz(T val) {
+  return val == 0;
+}
+template <typename T>
+bool WABT_VECTORCALL Eq(T lhs, T rhs) {
+  return lhs == rhs;
+}
+template <typename T>
+bool WABT_VECTORCALL Ne(T lhs, T rhs) {
+  return lhs != rhs;
+}
+template <typename T>
+bool WABT_VECTORCALL Lt(T lhs, T rhs) {
+  return lhs < rhs;
+}
+template <typename T>
+bool WABT_VECTORCALL Le(T lhs, T rhs) {
+  return lhs <= rhs;
+}
+template <typename T>
+bool WABT_VECTORCALL Gt(T lhs, T rhs) {
+  return lhs > rhs;
+}
+template <typename T>
+bool WABT_VECTORCALL Ge(T lhs, T rhs) {
+  return lhs >= rhs;
+}
+template <typename T>
+T WABT_VECTORCALL IntClz(T val) {
+  return Clz(val);
+}
+template <typename T>
+T WABT_VECTORCALL IntCtz(T val) {
+  return Ctz(val);
+}
+template <typename T>
+T WABT_VECTORCALL IntPopcnt(T val) {
+  return Popcount(val);
+}
+template <typename T>
+T WABT_VECTORCALL IntNot(T val) {
+  return ~val;
+}
+template <typename T>
+T WABT_VECTORCALL IntNeg(T val) {
+  return ~val + 1;
+}
+template <typename T>
+T WABT_VECTORCALL Add(T lhs, T rhs) {
+  return CanonNaN(lhs + rhs);
+}
+template <typename T>
+T WABT_VECTORCALL Sub(T lhs, T rhs) {
+  return CanonNaN(lhs - rhs);
+}
+template <typename T>
+T WABT_VECTORCALL IntAnd(T lhs, T rhs) {
+  return lhs & rhs;
+}
+template <typename T>
+T WABT_VECTORCALL IntOr(T lhs, T rhs) {
+  return lhs | rhs;
+}
+template <typename T>
+T WABT_VECTORCALL IntXor(T lhs, T rhs) {
+  return lhs ^ rhs;
+}
+template <typename T>
+T WABT_VECTORCALL IntShl(T lhs, T rhs) {
+  return lhs << ShiftMask(rhs);
+}
+template <typename T>
+T WABT_VECTORCALL IntShr(T lhs, T rhs) {
+  return lhs >> ShiftMask(rhs);
+}
+template <typename T>
+T WABT_VECTORCALL IntMin(T lhs, T rhs) {
+  return std::min(lhs, rhs);
+}
+template <typename T>
+T WABT_VECTORCALL IntMax(T lhs, T rhs) {
+  return std::max(lhs, rhs);
+}
+template <typename T>
+T WABT_VECTORCALL IntAndNot(T lhs, T rhs) {
+  return lhs & ~rhs;
+}
+template <typename T>
+T WABT_VECTORCALL IntAvgr(T lhs, T rhs) {
+  return (lhs + rhs + 1) / 2;
+}
+template <typename T>
+T WABT_VECTORCALL Xchg(T lhs, T rhs) {
+  return rhs;
+}
```

Comment:
Perhaps wrap these blocks of templates in `// clang-format off` `// clang-format on` ?

Reply:
Done